include/sdbus-c++/IProxy.h

```diff
@@ -89,7 +90,7 @@ namespace sdbus {
          *
          * @throws sdbus::Error in case of failure
          */
-        virtual AsyncMethodCall createAsyncMethodCall(const std::string& interfaceName, const std::string& methodName) = 0;
+        virtual AsyncMethodCall createAsyncMethodCall(const std::string& interfaceName, const std::string& methodName, uint64_t timeout = 0) = 0;
```

Comment:
Now that I think of it, sd-bus API is designed such that it doesn't ask for timeout upon message creation, but message dispatch action. Couldn't we model it the same way? So instead of `IProxy::createMethodCall`/`IProxy::createAsyncMethodCall`, we would add an optional timeout parameter in `IProxy::callMethod` virtual methods for both sync and async. (And for those we can of course create a std::chrono::duration non-virtual inline overload, which would cast to usec and delegate to virtual counterpart, in a similar way as proposed above.)

IMO this would make code changes smaller, and wouldn't require keeping timeout as member of `MethodCall`/`AsyncMethodCall`.